src/Bicep.Core/Emit/ForSyntaxValidatorVisitor.cs

```diff
@@ -60,17 +60,26 @@ public static bool IsAddingPropertyLoopAllowed(SemanticModel semanticModel, Obje
 
         public override void VisitResourceDeclarationSyntax(ResourceDeclarationSyntax syntax)
         {
+            // This check is separate from IsLoopAllowedHere because this is about the appearance of a
+            // nested resource **inside** a loop.
+            if (this.semanticModel.Binder.GetNearestAncestor<ForSyntax>(syntax) is ForSyntax)
+            {
+                this.diagnosticWriter.Write(DiagnosticBuilder.ForPosition(syntax.Span).NestedResourceNotAllowedInLoop());
```

Comment:
I might be reading this wrong, but isn't this error message the wrong way round ("nested resource not allowed in a loop" vs "loop not allowed on nested resources")?

Reply:
This is meant to block the following:

```
resource a '...' = [for i in items: {

  // error here: A nested resource cannot appear inside of a resource with a for-expression
  resource b '...' = {
  }
}]
```

Hopefully this is clearer.